src/trio/testing/_raises_group.py

```diff
@@ -307,28 +325,60 @@ def __init__(
         self,
         exception: type[E] | Matcher[E] | E,
         *other_exceptions: type[E] | Matcher[E] | E,
-        strict: bool = True,
+        allow_unwrapped: bool = False,
+        flatten_subgroups: bool = False,
         match: str | Pattern[str] | None = None,
         check: Callable[[BaseExceptionGroup[E]], bool] | None = None,
+        strict: None = None,
     ):
         self.expected_exceptions: tuple[type[E] | Matcher[E] | E, ...] = (
             exception,
             *other_exceptions,
         )
-        self.strict = strict
+        self.flatten_subgroups: bool = flatten_subgroups
+        self.allow_unwrapped = allow_unwrapped
         self.match_expr = match
         self.check = check
         self.is_baseexceptiongroup = False
 
+        if strict is not None:
+            warnings.warn(
+                DeprecationWarning(
+                    "`strict=False` has been replaced with `flatten_subgroups=True`"
+                    " with the introduction of `allow_unwrapped` as a parameter."
+                ),
+                stacklevel=2,
+            )
+            self.flatten_subgroups = not strict
+
+        if allow_unwrapped and other_exceptions:
+            raise ValueError(
+                "You cannot specify multiple exceptions with `allow_unwrapped=True.`"
+                " If you want to match one of multiple possible exceptions you should"
+                " use a `Matcher`."
+                " E.g. `Matcher(check=lambda e: isinstance(e, (...)))`"
+            )
+        if allow_unwrapped and isinstance(exception, RaisesGroup):
+            raise ValueError(
+                "`allow_unwrapped=True` has no effect when expecting a `RaisesGroup`."
+                " You might want it in the expected `RaisesGroup`, or"
+                " `flatten_subgroups=True` if you don't care about the structure."
+            )
```

Comment:
Another bad construction: `allow_unwrapped` and `match`. Currently the behavior is incorrect (it ignores `match` if there's no exception group).

Reply:
I think the current behaviour is somewhat sensible: "match the text of the group if it's in a group, if you want to match the text of the exception you should use a matcher object".
Same thing for `check`.

But I could see that being unexpected, and we're better off banning `allow_unwrapped` plus any of `match`/`check`. And in the error message instructing users to use a `Matcher` object if they intended to `match`/`check` on the exception; or do extra checks afterwards on the group if they wanted constraints on the group. The latter should be sufficiently rare that it's fine to force users to jump through hoops for it.